daemon/api_systems_test.go

```diff
@@ -320,6 +320,9 @@ func (s *systemsSuite) TestSystemActionRequestWithSeeded(c *check.C) {
 	bootloader.Force(bt)
 	defer func() { bootloader.Force(nil) }()
 
+	r := boot.EnableTestingRebootFunction()
+	defer r()
```

Comment:
I see, I would expect this to mock boot.Reboot instead

Reply:
Got it, so in the end I've changed things so `EnableTestingRebootFunction()` is used only inside the boot package, and now I mock the calls to `boot.Reboot()` in the daemon package so no call to shutdown is done in those tests, leaving testing that bit to the boot package.